packages/kit/src/vite/utils.js

```diff
@@ -257,6 +260,14 @@ export function prevent_illegal_vite_imports(node, illegal_imports, out_dir) {
 	if (chain) throw new Error(format_illegal_import_chain(chain, out_dir));
 }
 
+/*
+ * Vite does some weird things with import trees in dev
+ * for example, a Tailwind app.css will appear to import
+ * every file in the project. This isn't a problem for
+ * Rollup during build.
+ */
+const module_types = new Set(['.js', '.ts', '.svelte']);
```

Comment:
do we need .mts,.mjs as well? or could it be inverted to stop looking when it encounters a `.css` ? Or is there another boolean on the module graph to see if it is a css module?

This is risky. Right now there is no great support for custom extensions, but it might change and at that point .anything could be a svelte file.

Reply:
I had suggested inverting seems risky. What else might this happen with? Postcss, less, etc.? I don't think we should ever support giving a different extension to a `.svelte` file. That just seems weird

Reply:
mdsvex with `.svx` or `.md` would like a word

Reply:
at the end of the day this is all tailwinds fault btw. it should not declare a dependency on a private env file in the first place

Reply:
looks like you can add configuration to tailwind to avoid this: https://github.com/sveltejs/kit/issues/5747#issuecomment-1199121656 

imho that's better than adding this restriction that might miss something.

Reply:
Is there a reason to be telling Tailwind JIT to look at JS files anyway? Is that a recommendation on their website?

I agree that updating `tailwind.config.cjs` in a project seems better than adding this escape hatch here in Kit.

Reply:
> Is there a reason to be telling Tailwind JIT to look at JS files anyway? Is that a recommendation on their website?

That's the default config you get when setting up a new Tailwind project. The reason for it is here: https://github.com/sveltejs/kit/issues/5747#issuecomment-1199546971

Reply:
Hmm, yeah, that's unfortunate. I'm not sure what I'd suggest here. I'm not saying I _never_ want to make a change within SvelteKit to accommodate this, but I do think we should revert that change from this PR and have it just be about the fairly uncontroversial Windows fix so we can get that out the door.

Reply:
I get the reticence here, but it seems better to go this route (with the extra file extensions, probably?) and fix a bug that _actually prevents using SvelteKit_ than to sit around and try to find some other upstream fix. We're still being fully strict in `build`, so at worst, users would be able to develop their app and then have to go back and adjust some imports when they try to build.

Reply:
Like, this issue is preventing _me_ from upgrading to use _the feature I implemented_, and there's no other way I can think of to fix it -- Tailwind really _is_ depending on those files during dev. This bug represents a complete blocker to using Tailwind in a SK project at all.

Reply:
Yeah, this seems like a pretty good compromise to me. Check `.js`/`.ts`/`.svelte` files during dev. Less common stuff like `.mdx` can get checked during build

Reply:
Okay, quick poll on which filetypes to consider. I think we should at least add `mts|mjs`. I don't really know why we wouldn't go ahead and add the three common mdsvex types: `svelte.md|svx|md`. That should cover virtually all bases. Anything else is such an edge case that we'll catch it in build and not have to feel bad about it.

Reply:
Hm. Yeah, I guess so. If we don't run into this sort of issue during `build` (i.e., we can continue to deny all imports of forbidden modules across the board), then during `dev` it's probably better to blacklist JS/TS/Svelte files than it is to whitelist CSS/Less/SASS/etc. files.

Reply:
Yeah, this is _strictly_ a dev-time limitation. I think it has something to do with HMR.

If we can get a quick consensus on the filetypes I'll make an update and it'll be good to go. I think the complete list should be:

```
ts
js
svelte
mts
mjs
cts
cjs
svelte.md
svx
md
```

That's a relatively short but (I think?) all-encompassing list that should avoid weirdness about things that aren't actually imports and provide as close to a buildtime experience as possible.

Reply:
I approve that list

Reply:
sounds fine to me